test/src/test/java/lib/layout/IconTest.java

```diff
@@ -151,7 +151,7 @@ public void testTasks() throws Exception {
         assertIconToSymbolOkay(taskDivs.get(5).getElementsByTagName("svg").get(0));
 
         assertIconToImageOkay(taskDivs.get(6).getElementsByTagName("img").get(0), "/plugin/xxx/icon.png");
-        assertIconToImageOkay(taskDivs.get(7).getElementsByTagName("img").get(0), "/plugin/xxx/icon.png");
```

Comment:
Based on the assertion failure, I think modifying the test is the wrong direction.

The assertion failure says:

```
Expected: a string ending with "/plugin/xxx/icon.png"
     but: was "/jenkinsplugin/xxx/icon.png"
```

If the Jenkins controller is configured to run with the prefix "/jenkins" (behind an nginx reverse proxy like mine is), then the value currently returned ("/jenkinsplugin/xxx/icon.png") seems incorrect.  I think it should be returning "/jenkins/plugin/xxx/icon.png".

Unfortunately, I don't know the cases where it should not be adding the leading slash.  Maybe the conditional that was deleted needs to be replaced with a check if the string starts with "image" or "plugin" and only then add the leading slash?

Reply:
> Based on the assertion failure, I think modifying the test is the wrong direction.

Yeah, I was quickly messing with things. Already reverted 👍🏻 

Functions#tryGetIconPath determines the icon source by either going off /images/ or /plugin/, but that check doesn't suffice for plugins which add the icon from an URL.
I've added an exemption for http, that suffices for plugins which read icons from foreign sources.